File: BlackJack.ViewModels/GameViews/GetAllStepsGameView.cs
using BlackJack.ViewModels.EnumViews;
using System;
using System.Collections.Generic;

namespace BlackJack.ViewModels.GameViews
{
    public class GetAllStepsGameView
    {
        public List<PlayerStepGetAllStepsGameViewItem> PlayerSteps { get; set; }

        public GetAllStepsGameView()
        {
            PlayerSteps = new List<PlayerStepGetAllStepsGameViewItem>();
        }
    }

    public class PlayerStepGetAllStepsGameViewItem
    {
        public Guid Id { get; set; }
        public SuiteTypeEnumView Suite { get; set; }
        public RankTypeEnumView Rank { get; set; }
        public PlayerGetAllStepsGameView Player { get; set; }
        public GameGetAllStepsGameView Game { get; set; }
    }

    public class PlayerGetAllStepsGameView
    {
        public string PlayerId { get; set; }
        public string UserName { get; set; }
        public decimal Balance { get; set; }
        public decimal Bet { get; set; }
    }

    public class GameGetAllStepsGameView
    {
        public Guid GameId { get; set; }
        public GameStateTypeEnumView GameState { get; set; }
    }
}

File: BlackJack.WEB/ClientApp/src/app/shared/entities/game.views/get-all-steps.game.view.ts
import { SuiteType } from '../../enums/suite-type';
import { RankType } from '../../enums/rank-type';
import { GameStateType } from '../../enums/game-state-type';


export class GetAllStepsGameView {
  playerSteps?: Array<PlayerStepGetAllStepsGameViewItem>;

  constructor() {
    this.playerSteps = new Array <PlayerStepGetAllStepsGameViewItem>();
  }
}

export class PlayerStepGetAllStepsGameViewItem {
  id?: string;
  suite?: SuiteType;
  rank?: RankType;
  player?: PlayerGetAllStepsGameView;
  game?: GameGetAllStepsGameView;

  constructor() {
    this.player = new PlayerGetAllStepsGameView();
    this.game = new GameGetAllStepsGameView();
  }
}

export class PlayerGetAllStepsGameView {
  playerId?: string;
  balance?: number;
  bet?: number;
}

export class GameGetAllStepsGameView {
  gameId?: string;
  gameState?: GameStateType;
}
